test/gurka/test_instructions_roundabout.cc

```diff
@@ -203,3 +203,72 @@ TEST_F(InstructionsRoundabout, RoundaboutExitSuppressed) {
                                                             "You have arrived at your destination.",
                                                             "");
 }
+
+/*************************************************************/
+TEST(InstructionsRoundaboutRegression, TurnChannelRoundaboutExitRegression) {
+  // Regression test for roundabout exit that is also a turn channel
+  // https://www.openstreetmap.org/query?lat=48.62346&lon=2.46777#map=18/48.62310/2.46813
+  const std::string ascii_map = R"(
+    K---L--------O---P
+        \       /
+         M     N
+         \    /
+    A     E--F
+    |    /    \
+    B---D      G---J
+    |    \    /
+    C     I--H
+
+  )";
+
+  const gurka::ways ways = {
+      {"ABC", {{"highway", "service"}}},
+      {"BD", {{"highway", "tertiary"}}},
+      {"DIHGFED", {{"highway", "secondary"}, {"junction", "roundabout"}, {"name", ""}}},
+      {"GJ", {{"highway", "secondary"}}},
+      {"KLOP", {{"highway", "trunk"}, {"oneway", "yes"}, {"name", "La Francilienne"}}},
+      {"LME",
+       {{"highway", "trunk_link"},
+        {"oneway", "yes"},
+        {"destination", "Corbeil-Esses-Centre; Z.I. de l'Apport Paris;Ports"},
+        {"junction:ref", "30"}}},
+      {"FNO", {{"highway", "trunk_link"}, {"oneway", "yes"}, {"name", ""}}},
+  };
+
+  const gurka::nodes nodes = {
+      {"L",
+       {{"highway", "motorway_junction"}, {"ref", "30"}, {"exit_to", "Corbeil-Essonnes centre"}}}};
+
+  const double gridsize = 10;
+  const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize, {5.1079374, 52.0887174});
+
+  auto map = gurka::buildtiles(layout, ways, nodes, {}, "test/data/gurka_roundabouts_regression",
+                               {{"mjolnir.admin",
+                                 {VALHALLA_SOURCE_DIR "test/data/netherlands_admin.sqlite"}}});
+
+  const std::string& request =
+      (boost::format(
+           R"({"locations":[{"lat":52.08862730752708,"lon":5.107829332318943},{"lat":52.08843613199833,"lon":5.109036862850189}],"costing":"auto"})"))
```

Comment:
@gknisely how did you determine the lat/lon of the request? Is it possible to specify the source & destination using their alphabet strings (and then use `map.nodes.at(from).lat())` & `map.nodes.at(from).lng())` ) instead? I think this src/dest in this test are `K -> P` ?

Reply:
@mandeepsandhu I updated with input points `1` and `2` to clarify.